handler.go

```diff
@@ -60,6 +63,49 @@ func (o Options) ReadOptions(key string, obj interface{}) error {
 	return json.Unmarshal(msg, obj)
 }
 
+// Stream expects an array of json elements and allows to process them sequentially
+// this helps when one needs to parse a large json without having any memory leaks.
+// Returns ErrEmpty on empty key or when there are no more elements.
+// Returns ErrState when the stream has finished/encountered a Decode error.
+func (o Options) Stream(key string) (func(obj interface{}) error, error) {
+	msg := o[key]
+	if len(msg) == 0 {
+		return nil, errors.Wrap(errors.ErrEmpty, "data")
+	}
+
+	dec := json.NewDecoder(bytes.NewReader(msg))
+
+	// read opening bracket
+	if _, err := dec.Token(); err != nil {
+		return nil, errors.Wrapf(errors.ErrInput, "opening bracket %s", err)
+
+	}
+
+	closed := false
+
+	return func(obj interface{}) error {
+		if closed {
+			return errors.Wrap(errors.ErrState, "closed")
```

Comment:
This could return the same error type as no more data does (`ErrEmpty` or `ErrIterationDone`). For the user it will be easier to handle, because it does not require a separate handling. And the API is simplified -- calling a done iterator function is safe and always returns the same result.

Reply:
I was thinking about it, the closed bit only happens when we returned an error already so it does not in fact require any handling unless the client does something wrong.
That is why I'd like to keep it as different error and be able to differentiate. `ErrState` shouldn't ever be reached by any client that does it's error handling right.